plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -146,6 +147,89 @@ fu_plugin_thunderbolt_is_host (GUdevDevice *device)
 	return g_str_has_prefix (name, "domain");
 }
 
+static gchar *
+fu_plugin_thunderbolt_get_force_power_path (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	GList *devices;
+	const gchar *basepath;
+	g_autofree gchar *built_path = NULL;
+
+	devices = g_udev_client_query_by_subsystem (data->udev, "wmi");
+	for (GList *l = devices; l != NULL; l = l->next) {
+		GUdevDevice *device = l->data;
+		basepath = g_udev_device_get_sysfs_path (device);
+		built_path = g_build_path ("/", basepath,
+					   "force_power", NULL);
+		if (g_file_test (built_path, G_FILE_TEST_IS_REGULAR))
+			break;
+		built_path = NULL;
+	}
+	g_list_foreach (devices, (GFunc) g_object_unref, NULL);
+	g_list_free (devices);
+
+	if (built_path)
+		return g_strdup (built_path);
+	return NULL;
+}
+
+static gboolean
+fu_plugin_thunderbolt_set_force_power (FuPlugin *plugin, gboolean enable,
+				       GError **error)
+{
+	g_autofree gchar *path = NULL;
+	gint fd;
+	gint ret;
+
+	path = fu_plugin_thunderbolt_get_force_power_path (plugin);
+	if (path == NULL) {
+		return FALSE;
```

Comment:
If we're returning FALSE we need to set a GError.

Reply:
In the way this commit was done this didn't work well (since it was an implicit test for support).  In the new plugin approach I moved this over to have an explicit test for support and set the GError in this path.